ddsrouter_event/include/ddsrouter_event/wait/CounterWaitHandler.hpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CounterWaitHandler.hpp
+ */
+
+#ifndef _DDSROUTEREVENT_WAIT_BOOLEANWAITHANDLER_HPP_
+#define _DDSROUTEREVENT_WAIT_BOOLEANWAITHANDLER_HPP_
+
+#include <atomic>
+#include <condition_variable>
+#include <mutex>
+
+#include <ddsrouter_event/wait/WaitHandler.hpp>
+
+namespace eprosima {
+namespace ddsrouter {
+namespace event {
+
+using CounterType = int32_t;
+
+/**
+ * @brief This WaitHandler allows to wait for an int value
+ *
+ * This works as an int value that allows to wait for a specific value or threshold.
+ */
+class CounterWaitHandler : protected WaitHandler<CounterType>
+{
+public:
+
+    /**
+     * @brief Construct a new Boolean Wait Handler object
+     *
+     * @param value to initialize internal value
+     * @param enabled whether the object starts enabled or disabled
+     */
+    CounterWaitHandler(
+            CounterType value,
```

Comment:
Set default ``value`` to be 0, as this is almost always the case.

Reply:
I am refactoring my mind about default arguments... I the internal value is a very important value of the object, and this the user should know what it is and what it wants to start with. Using a default value will only make the constructor call shorter, which I think is not the idea of default values.